src/main/java/com/minecolonies/coremod/commands/colonycommands/DoRaidTonightCommand.java

```diff
@@ -44,31 +52,50 @@ public String getCommandUsage(@NotNull final ICommandSender sender)
     }
 
     @Override
-    public void execute(@NotNull final MinecraftServer server, @NotNull final ICommandSender sender, @NotNull final String... args) throws CommandException
+    public void execute(@NotNull final MinecraftServer server, @NotNull final ICommandSender sender, @NotNull final List<ActionArgument> actionArgumentList,
+            @NotNull final Map<String, Object> argumentValueByActionArgumentNameMap) throws CommandException
     {
-        if (sender instanceof EntityPlayer && !isPlayerOpped(sender))
+        Colony colony = null;
+        final Object colonyObject = argumentValueByActionArgumentNameMap.get("colony");
+        if (null != colonyObject)
         {
-            sender.sendMessage(new TextComponentString("Must be OP to use command"));
-            return;
+            colony = (Colony) colonyObject;
         }
 
+        executeShared(server, sender, colony);
+    }
+
+    @Override
+    public void execute(@NotNull final MinecraftServer server, @NotNull final ICommandSender sender, @NotNull final String... args) throws CommandException
+    {
+        Colony colony = null;
         if (args.length != 0)
         {
-            final Colony colony = ColonyManager.getColony(Integer.parseInt(args[0]));
+            colony = ColonyManager.getColony(Integer.parseInt(args[0]));
             if (colony == null)
             {
                 sender.sendMessage(NO_COLONY_FOUND_MESSAGE_ID);
                 return;
             }
-
-            colony.getBarbManager().setWillRaidTonight(true);
-
-            sender.sendMessage(SUCCESSFUL);
         }
         else
         {
             sender.sendMessage(NO_ARGUMENTS);
         }
+
+        executeShared(server, sender, colony);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') "NullPointerException" will be thrown when invoking method "executeShared()". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2259)